src/main/java/bio/terra/service/resourcemanagement/AzureDataLocationSelector.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.service.resourcemanagement;
+
+import bio.terra.common.exception.NotImplementedException;
+import bio.terra.model.BillingProfileModel;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+@Primary
+@Component
+public class AzureDataLocationSelector {
+
+    public String storageAccountNameForDataset(String prefix, String datasetName, BillingProfileModel billingProfile) {
+        int maxStorageAccountNameLength = 24;
+        int randomLength = maxStorageAccountNameLength - prefix.length();
+        return prefix + armUniqueString(datasetName + billingProfile.toString(), randomLength);
+    }
+
+    /**
+     * Generate an n-character unique string in a way that mimics what azure's ARM template function uses.
+     * Note: this method is deterministic (e.g. calling with the same seed value will provide the same result)
+     * @param seed The value to generate a unique string for
+     * @param size The number of characters to generate
+     * @return An n-character unique string based on what was passed into seed
+     */
+    private String armUniqueString(String seed, int size) {
```

Comment:
This code appears to be duplicated in `AzureResourceConfigurationTest`

Reply:
Yeah, I'll pull out the one from the testing